### DISCUSSION TITLE
Welcome to AdvancedProfanityFilter Discussions!

### DISCUSSION THREAD

**Discussion by richardfrost** [FrostCo/AdvancedProfanityFilter#269] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by BE-Code** [comment]
Hi! I am looking to get involved helping with open source projects and this is one that I use all the time. I am pretty new to node/typescript but I do have a decent amount of experience with programming in general. So, I'm new to the project, but I am here to help if I can!

**Reply by richardfrost** [reply]
Thanks for offering to help out! If you have something specific you'd like to help out with let me know. Otherwise, feel free to look through open issues and see if one looks like you might be interested in helping out with it.

I'll also keep on the lookout for some good "first issues", which definitely do crop up pretty often. What other languages do you have experience with? Are you interested in doing front-end work (HTML/CSS), or would you rather stick to the "back-end" code? 

We'd love to have you on our discord server if you'd like to help support users that way.

**Reply by BE-Code** [reply]
Thank you! I have experience with C++, python, html/css, and a few others. I often look at webpage code with the browser's developer tools and I've written a lot of html and css, but very little javascript. One thing I am a little bit confused about is why does this project have a back end? Doesn't a chrome extension run client side?

**Reply by richardfrost** [reply]
Nice! That's a lot of good variety. When I say "backend" here, I was just referring to the base code, or where the logic resides. There is definitely not a backend or API that the project relies on. Everything happens client-side. I was trying to refer to 2 different aspects of the codebase:
- Front-end: Options page and Popup page
- Back-end (Logic): Code that runs to actually do the filtering

**Reply by BE-Code** [reply]
Got it! I'd love to do both aspects.

**Reply by richardfrost** [reply]
I just got an interesting request to add a way to remove infrequently matched words from the filter. If you have stats enabled you can go to Option page > Stats to see the breakdown. One user has hundreds of words/phrases that they have added and now they've bumped into the limit for syncing. One solution might be to allow users to move the word/phrase storage to local storage which has a much higher limit. But the user also suggested perhaps adding a feature that could remove words under a threshold (probably an input box).

The reason I bring it up is the automated word removal might be a good place to start contributing if you'd like. Basically I imagine an text input box on the stats page and a button that would become active when a number is placed in that text box. Any word that has less than that number would then be removed from the config when that button is pushed. If you're interested or have any other questions let me know. Also, be sure to read through [CONTRIBUTING.md](https://github.com/richardfrost/AdvancedProfanityFilter/blob/master/CONTRIBUTING.md) and that you agree with it before contributing.

**Reply by richardfrost** [reply]
Sorry, I forgot to tag you @BE-Code in that last message. If you'd rather do something else, I'm sure there will be other tasks as well. Thanks!

**Reply by BE-Code** [reply]
Sorry! I saw your message, but school got really busy. Thank you for finding something I can work on though! How do I run the code after I've made a change to test it?

**Reply by richardfrost** [reply]
You're totally fine. Life has been crazy busy for me the last couple weeks as well. Here is a quick guide I threw together some time ago: https://github.com/richardfrost/AdvancedProfanityFilter/wiki/Development

Once you have Node.js installed, you should probably fork the repo, then clone that repo using `git`, and in that directory run `npm install`. That should take care of installing all the dev dependencies for you. Then you can start your development, and when you are ready to test your changes, run `npm run build` and then load an unpacked extension using your browser to load that version of the extension. If you then make another change, re-run the build command and then refresh/reload the extension in your browser to get the latest code.

Let me know if you have any questions!

**Reply by richardfrost** [reply]
@BE-Code Just wanted to let you know that I've built out the functionality to remove less-used words from the Stats page now. If you'd like to see the work, you can find it in #352. I'll see if I can find a few other projects that might be a good intro and let you know. Let me know if you have any questions on anything. Thanks!

**Comment by WolfD1g1tal** [comment]
Hello, my name is wolf. I had a question about your app i can no longer go without. it says that peacock is in the list of domains but it is not engaging. turning green or filtering audio. it's not even hiding the cc. is it not working or am i doing something wrong? also any chance you will add shudder?

**Reply by richardfrost** [reply]
Hi @WolfD1g1tal, it does look like Peacock has updated their site in regards to the captions. I think I have worked out the updates and will get it included in the next version of APF, but if you'd like to test it now here it is:

```json
{
  "www.peacocktv.com": [
    {
      "displaySelector": "div.video-player__subtitles",
      "mode": "elementChild",
      "parentSelector": "div.video-player__subtitles > span.video-player__subtitles__cue span.video-player__subtitles__line",
      "simpleUnmute": true,
      "tagName": "#text"
    },
    {
      "className": "video-player__subtitles",
      "displaySelector": "div.video-player__subtitles",
      "mode": "element",
      "subtitleSelector": "span.video-player__subtitles__cue > span.video-player__subtitles__line",
      "tagName": "DIV"
    }
  ]
}
```
(This goes in the "Custom Sites" area of the Options page > Audio tab. Once its in there, click the blue "Save" button underneath and refresh your Peacock tab).

I'd love to hear if it works for you before including it in an update.

As for Shudder, are you meaning https://www.shudder.com/ ?

**Reply by richardfrost** [reply]
@WolfD1g1tal Just wanted to follow up again, is this the site you were referring to with the name "Shudder"? https://www.shudder.com/

**Comment by WolfD1g1tal** [comment]
Hey. Yea that is the one. its getting pretty big. You may also want to add https://moviesanywhere.com/explore  you prolly heard about it but its taking all the purchase to own sites and bringing them all together as well as place for the digital copies. It might be a good one to add as well. Ok I watched some stuff on peacock and it seemed to run just fine. I’d say its good to go. I also checked it with peacock plus so premium and peacock plus did great.  Thank you heaps!

**Comment by WolfD1g1tal** [comment]
Hey Richard. Needed to let you know that advanced mode suddenly stopped working properly on YouTube and peackock. I tried eliminating different extensions well I tried to disable yours first and both pages loaded.
I turned back on I got the same browser hang. I have to close it with task manager just like the rest of the times.
So next I tried leaving apf on and disabled ghostery. Hang
Enabled ghostery, disabled apf. No hang.
Enabled apf. Hang
Then I noticed it was working fine on hbo max with apf enabled. So I looked and noticed I had YouTube and peacock set to advanced. Hbo max was set to deep.
So I switched both hbo and tube to deep. No Hang.
I had been watching peackock most of the day with no troubles. Then suddenly this happened.
Chrome updated to current/windows 10
Might want to see if yours is having kittens as well.
I have other extensions I didn’t try turning off.

**Reply by richardfrost** [reply]
Thanks for reporting the issue @WolfD1g1tal, that's definitely strange. First off, I'd recommend only using "Advanced" and "Deep" sites that need it. That being said it seems to be working for me on the sites you mentioned. Have you added any new words recently? Its definitely possible that could be interfering with it. If you'd like, you can send me a copy of your config (which you can export from the Config tab), or make a backup, restore defaults, and see if advanced mode works after that.

Right now, none of the officially supported muting sites need anything but "Normal" mode.

**Comment by WolfD1g1tal** [comment]
Hey Richard.
Just letting you know I checked the shudder and movies anywhere and both are working well. The movies anywhere, I only have titles there from the Amazon prime library. However if movies anywhere is using it’s own playback software there should be no difference in movies from Vudu or vimeo or any other library because anywhere is using it’s own player right? Thanks the heads up on using normal for sites .
Can’t wait till you get them up in next build. I realized I couldn’t just paste the new code under the previous and add them all at the same time. Json no likey loll. I am perfectly fine copy n pasting when I want to watch. Ya rawk man. Ty for the speedy magic you do lol.

**Reply by richardfrost** [reply]
Thanks @WolfD1g1tal! That new update ([version 2.39.0](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/2.39.0)) should be available now, or very soon. In the meantime if you want to use both Shudder and Movies Anywhere you can combine them like this:

```json
{
  "moviesanywhere.com": [
    {
      "mode": "cue"
    }
  ],
  "www.shudder.com": [
    {
      "mode": "element",
      "subtitleSelector": "div.vjs-text-track-cue > div",
      "tagName": "DIV"
    }
  ]
}
```

For Movies Anywhere, I have movies synced from most of the platforms and they all seemed to work well with the player. Thanks again for the new site requests, and for helping to test them.

**Comment by winniecol** [comment]
**winniecol:** The app is no longer muting the profanities. It blinks in and out.  

> **richardfrost:** Could you share some more details about where you are using it? What device and browser? What streaming site?

**winniecol:** I’m using my laptop. My browser is Microsoft edge.

**Reply by richardfrost** [reply]
@winniecol Which streaming service isn't working for you?

**Reply by richardfrost** [reply]
> winniecol: Netflix.

@winniecol Sometimes APF's subtitle detection can get thrown off by changes to Netflix's subtitles appearance settings. Please try resetting them back to the defaults. To reset them, go to your Netflix account settings (https://www.netflix.com/YourAccount), expand your profile in the "Profile & Parental Controls" section and then choose to "change" Subtitle Appearance and then click on the "Reset to default" button.

After the subtitle appearance has been reset try the filter again.

**Reply by richardfrost** [reply]
> winniecol: Thank you!

Did that fix it for you @winniecol?

**Reply by richardfrost** [reply]
> winniecol: No, it didn't.

@winniecol Could you send me a copy of your extension's config? (Options page > Config tab > Export button). You can attach it here or send it in an email. Also, which country are you watching Netflix from?

**Reply by winniecol** [reply]
```json
{
  "audioWordlistId": 0,
  "censorCharacter": "*",
  "censorFixedLength": 1,
  "collectStats": true,
  "contextMenu": true,
  "customAudioSites": {
    "www.netflix.com": [
      {
        "className": "player-timedtext-text-container",
        "mode": "element",
        "note": "Fallback compatibility",
        "subtitleSelector": "span > span",
        "tagName": "DIV"
      },
      {
        "className": "player-timedtext-text-container",
        "mode": "element",
        "subtitleSelector": "span",
        "tagName": "DIV"
      }
    ]
  },
  "darkMode": null,
  "defaultSubstitution": "censored",
  "defaultWordMatchMethod": 0,
  "defaultWordRepeat": 1,
  "defaultWordSeparators": 1,
  "domains": {},
  "enabledDomainsOnly": true,
  "enabledFramesOnly": false,
  "fillerAudio": "",
  "filterMethod": 1,
  "filterWordList": true,
  "iWordAllowlist": [],
  "loggingLevel": 2,
  "muteAudio": true,
  "muteAudioOnly": true,
  "muteCueRequireShowing": false,
  "muteMethod": 1,
  "preserveCase": true,
  "preserveFirst": true,
  "preserveLast": false,
  "showCounter": true,
  "showSubtitles": 1,
  "showSummary": true,
  "showUpdateNotification": true,
  "substitutionMark": true,
  "syncLargeKeys": true,
  "wordAllowlist": [],
  "wordSubSeparator": ";;",
  "wordlistId": 0,
  "wordlists": [
    "Wordlist 1",
    "Wordlist 2",
    "Wordlist 3",
    "Wordlist 4",
    "Wordlist 5",
    "Wordlist 6"
  ],
  "wordlistsEnabled": true,
  "words": {
    "asshole": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 3,
      "repeat": 1,
      "separators": 1,
      "sub": "jerk"
    },
    "badass": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 1,
      "sub": "cool"
    },
    "bastard": {
      "case": 1,
      "lists": [],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 0,
      "sub": "idiot"
    },
    "bitch": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 0,
      "sub": "bench"
    },
    "cocksucker": {
      "case": 1,
      "lists": [],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 1,
      "sub": "suckup"
    },
    "cunt": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 0,
      "sub": "expletive"
    },
    "dammit": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 1,
      "sub": "dangit"
    },
    "damn": {
      "case": 1,
      "lists": [],
      "matchMethod": 1,
      "repeat": 0,
      "separators": 0,
      "sub": "dang"
    },
    "dumbass": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 1,
      "sub": "idiot"
    },
    "fag": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 0,
      "sub": "gay"
    },
    "faggot": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 0,
      "sub": "gay"
    },
    "fags": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 0,
      "sub": "gays"
    },
    "fuck": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 1,
      "sub": "freak"
    },
    "goddammit": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 2,
      "repeat": 1,
      "separators": 1,
      "sub": "dangit"
    },
    "hell": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 1,
      "sub": "heck"
    },
    "jackass": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 1,
      "sub": "jerk"
    },
    "nigga": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "bruh"
    },
    "nigger": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "man"
    },
    "niggers": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 1,
      "sub": "people"
    },
    "piss": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "pee"
    },
    "pissed": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "ticked"
    },
    "pussies": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "softies"
    },
    "pussy": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "softie"
    },
    "shit": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "crap"
    },
    "shit fuck bitch ass": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 0,
      "separators": 1,
      "sub": ""
    },
    "slut": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "tramp"
    },
    "tits": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "chest"
    },
    "twat": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 0,
      "repeat": 1,
      "separators": 0,
      "sub": "dumbo"
    },
    "twats": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 0,
      "sub": "dumbos"
    },
    "whore": {
      "case": 1,
      "lists": [
        1,
        2,
        3,
        4,
        5,
        6
      ],
      "matchMethod": 1,
      "repeat": 1,
      "separators": 0,
      "sub": "tramp"
    }
  },
  "youTubeAutoMatchCensored": true,
  "youTubeAutoSubsMax": 1,
  "youTubeAutoSubsMin": 1
}
```

**Reply by richardfrost** [reply]
@winniecol I finally got around to testing the config you provided, and it worked just fine for me on Netflix. Do you have any other extensions running that might be interfering with it? One thing you might do to isolate it is to allow APF to run in incognito/private mode, but no other extensions, and then try watching a video in there. There hasn't been any changes I'm aware of for Netflix, and it does seem to still be working with APF overall, so maybe something changed with your browser (I'm assuming it did work for you before based on your opening comment).

**Comment by winniecol** [comment]
Hello, Mr. Richard,
I have a request. Can more sites be added to the Advanced Profanity Filter? Specifically, AnimeSuge and FlixHd.

**Reply by richardfrost** [reply]
@winniecol since I can't verify the validity of these sites, I won't be able to add them as included/supported sites.

That said, you can try adding these configs and see if they work for you:

- [animesuge](https://github.com/FrostCo/AdvancedProfanityFilter/wiki/Audio-Muting:-Additional-Sites#site-animesugeto)
- [flixhd](https://github.com/FrostCo/AdvancedProfanityFilter/wiki/Audio-Muting:-Additional-Sites#site-flixhdcc)

To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration. Let me know if you need help combining them, or feel free to try them one at a time and let me know if they work.

**Comment by winniecol** [comment]
Thank you so much! It worked.
________________________________
